rclpy/rclpy/callback_groups.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from threading import Lock
+import weakref
+
+
+class CallbackGroup:
+    """Control when callbacks are allowed to be executed."""
+
+    def __init__(self):
+        super().__init__()
```

Comment:
`super` should be called with two arguments: the class type and `self`: `super(CallbackGroup, self).__init__()`

Reply:
Actually [PEP 3135](https://www.python.org/dev/peps/pep-3135/) seems to advocate to call it without arguments in Python 3. I remember that at least in the past with Python 2 that was "not a good idea". Our existing code passes the class name and `self`. Maybe we should update our code base to this new style?

Reply:
Looks like the PEP guarantees the old style will still work. To be honest I don't fully understand the advantages of the new style besides less typing. I can make a jira ticket to look into updating the existing code.